common/ledger/util/leveldbhelper/leveldb_provider.go

```diff
@@ -204,21 +204,20 @@ func (h *DBHandle) DeleteAll() error {
 	return nil
 }
 
+// NewUpdateBatch returns a new UpdateBatch that can be used to update the db
+func (h *DBHandle) NewUpdateBatch() *UpdateBatch {
+	return &UpdateBatch{
+		dbName: h.dbName,
+		Batch:  &leveldb.Batch{},
+	}
+}
+
 // WriteBatch writes a batch in an atomic way
 func (h *DBHandle) WriteBatch(batch *UpdateBatch, sync bool) error {
-	if len(batch.KVs) == 0 {
+	if batch == nil || batch.Len() == 0 {
```

Comment:
Not related to this PR. I was checking whether `leveldb.WriteBatch` returns err when the batch is nil (it does). You might be aware of the following. Then, I found that they use transaction (`OpenTransaction()`, `Commit()`) when the batch size is large to skip writing to journal. I remember once you said that we need to look at disabling the journal writes as we have a block store.